src/__tests__/Dataset-test.ts

```diff
@@ -379,14 +380,77 @@ describe("Dataset", function () {
     });
 
     it("add an asset", async function () {
-      expect((await testDs.uploadAsset(getDataDir("test102.nt"), "test102.nt")).getInfo().versions.length).to.equal(1);
+      expect(
+        (await testDs.uploadAsset(getDataDir("test102.nt"), {
+          mode: "throw-if-exists",
+          name: "test102.nt",
+        }))!.getInfo().versions.length,
+      ).to.equal(1);
+      let assetCount = 0;
+      for await (let asset of testDs.getAssets()) asset && assetCount++;
+      expect(assetCount).to.equal(1);
+    });
+    it("mode append-version", async function () {
+      expect(
+        (await testDs.uploadAsset(getDataDir("test102.nt"), { mode: "append-version", name: "test102.nt" })).getInfo()
+          .versions.length,
+      ).to.equal(1);
+      expect(
+        (await testDs.uploadAsset(getDataDir("test102.nt"), { mode: "append-version", name: "test102.nt" })).getInfo()
+          .versions.length,
+      ).to.equal(2);
+      expect(
+        (await testDs.uploadAsset(getDataDir("test102.nt"), { mode: "append-version", name: "test102.nt" })).getInfo()
+          .versions.length,
+      ).to.equal(3);
+
       let assetCount = 0;
       for await (let asset of testDs.getAssets()) asset && assetCount++;
       expect(assetCount).to.equal(1);
     });
+    it("throw-if-exists", async function () {
+      expect(
+        (await testDs.uploadAsset(getDataDir("test102.nt"), { mode: "throw-if-exists", name: "test102.nt" })).getInfo()
+          .versions.length,
+      ).to.equal(1);
+      await expect(
+        testDs.uploadAsset(getDataDir("test102.nt"), { mode: "throw-if-exists", name: "test102.nt" }),
+      ).to.be.rejectedWith(TriplyDbJsError, "but an asset with that name already exists.");
+    });
+    it("undefined mode", async function () {
+      expect((await testDs.uploadAsset(getDataDir("test102.nt"), "test102.nt")).getInfo().versions.length).to.equal(1);
```

Comment:
@dionenak Let's use the new interface throughout the tests, making it easier to propagate the correct code.

Reply:
@ivan-kovatchev I thought about it, but I decided to keep other tests as they are since we still have the old interface in many places. We propagate the use of the newest interface by putting this overload first for the dev to see.

Reply:
> (...) we still have the old interface in many places. 

Then we should change those places as well.

Reply:
The only reason we didn't remove the old interface was to not have a major release for this small piece of functionality. Eventually I can see us removing it entirely, as this is what we've done in the past.

Reply:
I didn't think that we were going to change the interface of this function, in core for example, right now.